src/Compilers/CSharp/Portable/FlowAnalysis/VariablesDeclaredWalker.cs

```diff
@@ -77,7 +77,12 @@ private void NoteDeclaredPatternVariables(BoundPattern pattern)
             if (IsInside && pattern.Kind == BoundKind.DeclarationPattern)
             {
                 var decl = (BoundDeclarationPattern)pattern;
-                _variablesDeclared.Add(decl.LocalSymbol);
+                if (decl.Variable.Kind == SymbolKind.Local)
```

Comment:
You didn't modify this file because the relevant APIs were never implemented for deconstruction. I'll submit a bug with a test case.

---
In reply to: [80756845](https://github.com/dotnet/roslyn/pull/13999#discussion_r80756845) [](ancestors = 80756845,80753230)